### PR TITLE
fix issue [fvt]2.13.4 nodeset osimage have no right output if command wrong #2971

### PR DESCRIPTION
**Pull request by immarvin**
fix issue: https://github.com/xcat2/xcat-core/issues/2971

complain and return if the argument is invalid, i.e, ``nodeset xxx osimage= [xxxx]``  or ``osimage= [xxxx]``in the chain

````
[root@c910f03c05k21 ~]# nodeset c910f03c05k27 osimage= rhels7.3-ppc64le-netboot-compute
Error: Invalid argument: 'osimage=rhels7.3-ppc64le-install-compute xxxx'
Usage: nodeset <noderange> [shell|boot|runcmd=bmcsetup|iscsiboot|osimage[=<imagename>]|offline]
[root@c910f03c05k21 perl]# echo $?
1
````

### CODE REVIEW

**Comment by robin2008** [comment]
@immarvin 
why not handle the CLI parameters checking in the first plugins, like xnba, petitboot, etc.
It is better to complain the option error, especially for the option format error,  as early as possible then we will not go deeper in business.

**Comment by immarvin** [comment]
hi @robin2008 ,code updated

File: perl-xCAT/xCAT/Usage.pm [hunk 1]
```diff
@@ -620,3 +620,38 @@ sub parseCommand {
...
+
+sub validateArgs {
+    my $command=shift; 
+    if ($command =~ /xCAT::Usage/) { $command = shift; }
+   
+    my @extrargs=@_;
+
+    if($command =~ m/^(nodeset|rinstall|winstall)$/ ){
+        foreach(@extrargs){
+            if($_ !~ m/^-[-]?\S+/){
```
Comment (robin2008) [inline perl-xCAT/xCAT/Usage.pm]: So we know our commands does not allow "-p foo" and "--port 12345",  better to describe the limitation in func header.

File: perl-xCAT/xCAT/Usage.pm [hunk 2]
```diff
@@ -620,3 +620,38 @@ sub parseCommand {
...
+                $count+=1;
+            }
+        }
+        if ($count!=1) {
+           return [1,"Invalid argument: '".join(" ",@extrargs)."'"];
```
Comment (robin2008) [inline perl-xCAT/xCAT/Usage.pm]: As this error is very straightforward, we can make the error message more descriptive,  just like: 
"Invalid arguments: check if any space is using by wrong in 'xxx'"

File: xCAT-server/lib/xcat/plugins/grub2.pm [hunk 3]
```diff
@@ -426,10 +426,13 @@ sub preprocess_request {
         return;
     }
 
-    if (@ARGV == 0) {
+   my $ret=xCAT::Usage->validateArgs($command,@ARGV);
```
Comment (robin2008) [inline xCAT-server/lib/xcat/plugins/grub2.pm]: It may not handle @ARGV == 0 case

Comment (immarvin) [inline xCAT-server/lib/xcat/plugins/grub2.pm]: if ``scala @argv==0`` ,  validateArgs will run into
````
 if ($count!=1) {
 +           return [1,"Invalid argument: '".join(" ",@extrargs)."'"];
````

**Pull request merged**